Src/FluentAssertions/Specialized/ExceptionAssertions.cs

```diff
@@ -148,19 +143,24 @@ public ExceptionAssertions(IEnumerable<TException> exceptions)
             params object[] becauseArgs)
             where TInnerException : Exception
         {
-            WithInnerException<TInnerException>(because, becauseArgs);
-
-            TInnerException[] expectedExceptions = Subject
-                .Select(e => e.InnerException)
-                .OfType<TInnerException>()
-                .Where(e => e?.GetType() == typeof(TInnerException)).ToArray();
-
-            Execute.Assertion
-                .ForCondition(expectedExceptions.Any())
-                .BecauseOf(because, becauseArgs)
-                .FailWith("Expected inner {0}{reason}, but found {1}.", typeof(TInnerException), SingleSubject.InnerException);
+            var exceptionExpression = AssertInnerExceptionExactly(typeof(TInnerException), because, becauseArgs);
+            return new ExceptionAssertions<TInnerException>(exceptionExpression.Cast<TInnerException>());
+        }
 
-            return new ExceptionAssertions<TInnerException>(expectedExceptions);
+        /// <summary>
+        /// Asserts that the thrown exception contains an inner exception of the exact type <param name="innerException" /> (and not a derived exception type).
+        /// </summary>
+        /// <param name="because">
+        /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+        /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+        /// </param>
+        /// <param name="becauseArgs">
+        /// Zero or more objects to format using the placeholders in <paramref name="because" />.
+        /// </param>
+        public virtual ExceptionAssertions<Exception> WithInnerExceptionExactly(Type innerException, string because = null,
```

Comment:
Remove `virtual` here as well.
As a rule-of-thumb, when we review PRs we only point out a single occurrence of something that should be changed.

Reply:
Yes, thank you for the Patience. It is my first time contributing to anything.